src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,194 @@
-    Nova objects present in the original catalog and is complete down to magnitude
-    ~6.5, while the faintest included star has mag=7.96. :cite:p:`bright-star-catalog`
+
+def select_stars(stars, pointing=None, radius=None, magnitude_cut=None, band="Vmag"):
+    """
+    utility function to cut stars based on magnitude and/or location
 
     Parameters
     ----------
+    stars: astropy table
+        Table of stars, including magnitude and coordinates
     pointing: astropy Skycoord
-       pointing direction in the sky (if none is given, full sky is returned)
+        pointing direction in the sky (if none is given, full sky is returned)
     radius: astropy angular units
-       Radius of the sky region around pointing position. Default: full sky
+        Radius of the sky region around pointing position. Default: full sky
     magnitude_cut: float
-        Return only stars above a given magnitude. Default: None (all entries)
+        Return only stars above a given apparent magnitude. Default: None (all entries)
+    band: string
+        wavelength band to use for the magnitude cut options are V and B. Default: 'B' (all entries)
 
     Returns
     -------
     Astropy table:
-       List of all stars after cuts with names, catalog numbers, magnitudes,
-       and coordinates
+       List of all stars after cuts with same keys as the input table stars
     """
-    from ctapipe.utils import get_table_dataset
 
-    catalog = get_table_dataset("yale_bright_star_catalog5", role="bright star catalog")
-
-    starpositions = SkyCoord(
-        ra=Angle(catalog["RAJ2000"], unit=u.deg),
-        dec=Angle(catalog["DEJ2000"], unit=u.deg),
-        frame="icrs",
-        copy=False,
-    )
-    catalog["ra_dec"] = starpositions
-
-    if magnitude_cut is not None:
-        catalog = catalog[catalog["Vmag"] < magnitude_cut]
+    stars_ = deepcopy(stars)
+    if magnitude_cut:
+        try:
+            stars_ = stars_[stars_[band] < magnitude_cut]
+        except KeyError:
+            raise ValueError(
+                f"The requested catalogue has no compatible magnitude for the {band} band"
+            )
 
     if radius is not None:
-        if pointing is None:
+        if pointing:
+            stars_["separation"] = stars_["ra_dec"].separation(pointing)
+            stars_ = stars_[stars_["separation"] < radius]
+        else:
             raise ValueError(
                 "Sky pointing, pointing=SkyCoord(), must be "
                 "provided if radius is given."
             )
-        separations = catalog["ra_dec"].separation(pointing)
-        catalog["separation"] = separations
-        catalog = catalog[separations < radius]
 
-    catalog.remove_columns(["RAJ2000", "DEJ2000"])
+    return stars_
+
+
+def get_star_catalog(catalog, magnitude_cutoff=8.0, row_limit=1000000):
+    """
+    Utility function to download a star catalog for the get_bright_stars function
+
+    Parameters
+    ----------
+    catalog: string
```

Comment:
Should update this to `catalog: str | StarCatalog` and maybe also add a type hint using the same, since that will make checking nicer. 

```suggestion
def get_star_catalog(catalog: StarCatalog | str, magnitude_cutoff=8.0, row_limit=1000000):
    """
    ...
    catalog: StarCatalog | str
    """
```

Note that fancy language servers like PyCharm actually parse this parameters section and will complain that  "string" is not a python type.

Reply:
At least currently, giving the enum does not work. You have to give the name of the enum member as string.

Reply:
See https://github.com/cta-observatory/ctapipe/pull/2625/files#r1816939901